opensearch/bbuy_queries.json

```diff
@@ -1,5 +1,34 @@
 {
   "settings": {
     "index.refresh_interval": "5s"
+  },
+  "mappings": {
+    "dynamic": "false",
+    "properties": {
+      "user": {
+        "type": "keyword"
+      },
+      "sku": {
+        "type": "keyword"
+      },
+      "category": {
+        "type": "keyword"
+      },
+      "query": {
+        "type": "text",
+        "analyzer": "english",
+        "fields": {
+          "keyword": {
+            "type": "keyword"
+          }
+        }
+      },
+      "click_time": {
+        "type": "date"
+      },
+      "query_time": {
+        "type": "date"
```

Comment:
As you have time in the data, I think it's a better idea to keep it by using the datetime type with a custom format.
I'm speaking hypothetically here as this is out of the scope of the current course or task, but in a real-world scenario, it would be useful if you wanna build reports based on the data or further enhance your search results based on behavior.

Reply:
Actually, I did not found a `datetime` datatype:
https://www.elastic.co/guide/en/elasticsearch/reference/7.17/date.html

Only `date` and `date_nanos` exists in ES from what I know.

Reply:
Actually, sorry, my mistake. There isn't a datetime.
I just had trouble with indexing the date-time in some records as the time didn't match the default expected format in ES, but that was last week and for some reason my memory betrayed me. 😄I've written the comment without checking what I did to solve it. 
Anyway, what I did is set more date formats to cover more date strings like so:
`"format": "yyyy-MM-dd HH:mm:ss.SSS||ordinal_date_time||strict_date_optional_time||epoch_millis"`

Reply:
Ah no problem.
I use the date filter of Logstash:
https://github.com/tomsquest/search_with_machine_learning_course/pull/1/files#diff-cc9f0cb81f77e90ca1034b54619a84daac3cdd2ef125a741a475478ca0090fbfR17

Kudo to my coworker, Jesse Peixoto (that also do the training) for that tips.